components-core/src/main/java/com/airwallex/android/core/model/Options+AirwallexHttpRequest.kt

```diff
@@ -0,0 +1,155 @@
+package com.airwallex.android.core.model
+
+import android.net.Uri
+import com.airwallex.android.core.AirwallexApiRepository.Companion.confirmPaymentIntentUrl
+import com.airwallex.android.core.AirwallexApiRepository.Companion.continuePaymentIntentUrl
+import com.airwallex.android.core.AirwallexApiRepository.Companion.createPaymentConsentUrl
+import com.airwallex.android.core.AirwallexApiRepository.Companion.createPaymentMethodUrl
+import com.airwallex.android.core.AirwallexApiRepository.Companion.disablePaymentConsentUrl
+import com.airwallex.android.core.AirwallexApiRepository.Companion.retrieveAvailablePaymentMethodsUrl
+import com.airwallex.android.core.AirwallexApiRepository.Companion.retrieveBanksUrl
+import com.airwallex.android.core.AirwallexApiRepository.Companion.retrievePaymentIntentUrl
+import com.airwallex.android.core.AirwallexApiRepository.Companion.retrievePaymentMethodTypeInfoUrl
+import com.airwallex.android.core.AirwallexApiRepository.Companion.trackerUrl
+import com.airwallex.android.core.AirwallexApiRepository.Companion.verifyPaymentConsentUrl
+import com.airwallex.android.core.AirwallexPlugins
+import com.airwallex.android.core.http.AirwallexHttpRequest
+import java.util.*
+
+@Suppress("LongMethod")
+fun Options.toAirwallexHttpRequest(): AirwallexHttpRequest {
+    val url = getUrl()
+
+    return when (this) {
+        is Options.ConfirmPaymentIntentOptions -> AirwallexHttpRequest.createPost(
```

Comment:
I wonder if all these same body ones can be merged by using a comma `,`

Reply:
No, as the property `request` is tied to some specific `Options` class unless we write a interface those `Options` conform to force it have some common fields.